forum/qa-plugin/q2a-change-username-limit/page/account.php

```diff
@@ -0,0 +1,548 @@
+        )['username_change_history'], true);
+
+    $history[] = [
+        'old' => $oldhandle,
+        'new' => $newhandle,
+        'date' => $date
+    ];
+
+    qa_db_query_sub('UPDATE ^users SET username_change_history=$ WHERE userid=#', json_encode($history), $userid);
+}
+
+require_once QA_INCLUDE_DIR.'db/users.php';
+require_once QA_INCLUDE_DIR.'app/format.php';
+require_once QA_INCLUDE_DIR.'app/users.php';
+require_once QA_INCLUDE_DIR.'db/selects.php';
+require_once QA_INCLUDE_DIR.'util/image.php';
+
+
+//	Check we're not using single-sign on integration, that we're logged in
+
+if (QA_FINAL_EXTERNAL_USERS) {
+    qa_fatal_error('User accounts are handled by external code');
+}
+
+$userid=qa_get_logged_in_userid();
+
+if (!isset($userid)) {
+    qa_redirect('login');
+}
+
+
+//	Get current information on user
+
+list($useraccount, $userprofile, $userpoints, $userfields)=qa_db_select_with_pending(
+    qa_db_user_account_selectspec($userid, true),
+    qa_db_user_profile_selectspec($userid, true),
+    qa_db_user_points_selectspec($userid, true),
+    qa_db_userfields_selectspec()
+);
+
+$changehandle=qa_opt('allow_change_usernames');
+$doconfirms=qa_opt('confirm_user_emails') && ($useraccount['level']<QA_USER_LEVEL_EXPERT);
+$isconfirmed=($useraccount['flags'] & QA_USER_FLAGS_EMAIL_CONFIRMED) ? true : false;
+$haspassword=isset($useraccount['passsalt']) && isset($useraccount['passcheck']);
+$isblocked = qa_user_permit_error() !== false;
+
+//	Process profile if saved
+
+// If the post_max_size is exceeded then the $_POST array is empty so no field processing can be done
+if (qa_post_limit_exceeded()) {
+    $errors['avatar'] = qa_lang('main/file_upload_limit_exceeded');
+}
+else {
+    if (qa_clicked('dosaveprofile') && !$isblocked) {
+        require_once QA_INCLUDE_DIR.'app/users-edit.php';
+
+        $inhandle = $changehandle ? qa_post_text('handle') : $useraccount['handle'];
+        $inemail = qa_post_text('email');
+        $inmessages = qa_post_text('messages');
+        $inmessagesemail = (int)qa_post_text('messages_email');
+        $inactivityemail = (int)qa_post_text('activity_email');
+        $inwallposts = qa_post_text('wall');
+        $inmailings = qa_post_text('mailings');
+        $inavatar = qa_post_text('avatar');
+
+        $inprofile=array();
+        foreach ($userfields as $userfield) {
+            $inprofile[$userfield['fieldid']] = qa_post_text('field_' . $userfield['fieldid']);
+        }
+
+        if (!qa_check_form_security_code('account', qa_post_text('code'))) {
+            $errors['page'] = qa_lang_html('misc/form_security_again');
+        }
+
+        else {
+            $errors = qa_handle_email_filter($inhandle, $inemail, $useraccount);
+
+            if (!isset($errors['handle'])) {
+                if (null !== checkLastChangeUsername($userid) && checkLastChangeUsername($userid) < 30) {
```

Comment:
IMO czytelniej będzie w ten sposób + mniejsza liczba wywołań funkcji.

```php
$lastTimeUsernameChanged = checkLastChangeUsername($userid);

if (-1 !== $lastTimeUsernameChanged && $lastTimeUsernameChanged < CHANGE_USERNAME_LIMIT_IN_DAYS) {
```

Reply:
Do tego funkcja `checkLastChangeUsername` mogłaby zwracać `0` lub `-1` zamiast `null`. Skoro w pozytywnym case zwraca liczbę, to w negatywnym też może ją zwracać zamiast `null`, żeby był jednolity typ.

Reply:
null jest wygodniejszy w obsłudze i czytelniejszy. Jeśli chodzi o wydzielenie do zmiennej to już to zrobiłem